### PR TITLE
Limit libarchive format to what we use

### PR DESCRIPTION
**Pull request by santrancisco**
We are only using libarchive for [7zip](https://github.com/ClickHouse/ClickHouse/blob/cba5921270247f4bb3275204159155bcf498219f/src/IO/Archives/createArchiveReader.cpp#L59)  and some [tar archive](https://github.com/ClickHouse/ClickHouse/blob/cba5921270247f4bb3275204159155bcf498219f/src/IO/Archives/createArchiveReader.cpp#L51C51-L51C66) formats. 

This change is to narrow down the support to just the one that we need in our code:

```
        ".tar"
        ".tar.gz"
        ".tgz"
        ".tar.zst"
        ".tzst"
        ".tar.xz"
        ".tar.bz2"
        ".7z"
```

By doing so, we reduce our attack surfaces in other file format included in libarchive by default (rar, xar, cab, etc..). 

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 609a60072f98ef710bd16692ae1203de4c68f71e with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60203/609a60072f98ef710bd16692ae1203de4c68f71e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60203/609a60072f98ef710bd16692ae1203de4c68f71e/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Good, although `rar` is nice to have.

**Review by antonio2368 — APPROVED** [review]
Thanks for fixing it!

**Comment by santrancisco** [comment]
@alexey-milovidov  - words on the street that rar format may have a few 0day :trollface:

**Comment by santrancisco** [comment]
Fast test failed ... `libarchive library is disabled` <-- oh this sucks... I guess it's not compiled for these test binaries? 🤔 Is there any reason it is disabled ? What do we normally do in this case ?

File: tests/queries/0_stateless/02994_libarchive_compression.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
```
Comment (antonio2368) [inline tests/queries/0_stateless/02994_libarchive_compression.sh:1]:
```suggestion
#!/usr/bin/env bash
# Tags: no-fasttest
```

fast tests just compile binary without any libs so we simply disable tests for it which do need some of the libs

**Pull request merged**